### ISSUE TITLE
ofxAVFoundationVideoPlayer - replacement for iOS and OSX video player

### ISSUE THREAD

**Issue by julapy** [openframeworks/openFrameworks#3445]
ive been working on ofxAVFoundationVideoPlayer as a replacement for iOS and OSX video player.
https://github.com/julapy/ofxAVFoundationVideoPlayer

the back story here is that, initially it was a port of ofxiOSVideoPlayer which i was able to make work across iOS and OSX thanks to the AVFoundation framework being designed to work the same across the two platforms. ive also been comparing it with ofAVFoundationPlayer (the current OSX video player) and making sure the functionality is the same and also borrowing some code from there to combine the best of the two video players.

in short, ofxAVFoundationVideoPlayer runs on iOS and OSX and completes some missing features on OSX.

@prisonerjohn would be really good if you could have a look at the code, run the examples and see if theres anything missing or that needs to change.
